test/scripts/run_test.py

```diff
@@ -47,47 +48,51 @@ def analyse_instrumentation_test_result(result):
     return ok_num
 
 
-def analyse_unit_test_result(result):
+def analyse_unit_test_result(result, mode, module):
+    xml_file_name = module + '-' + mode.value + '-result-' + XML_SUFFIX
     total_num = 0
-    fail_num = 0
+    success_num = 0
+    p = re.compile(".*?XML test result file generated at (.*?)\. Total tests (\d+).*?passed (\d+)")
     with open(result, 'r') as f:
         for line in f:
-            m_failure = re.match('Tests run:\s+(\d+).*?Failures:\s+(\d+)', line)
-            m_succeed = re.match('OK.*?(\d+) test', line)
-            if m_failure is not None:
-                total_num = int(m_failure.group(1))
-                fail_num = int(m_failure.group(2))
-            if m_succeed is not None:
-                total_num = int(m_succeed.group(1))
-    return total_num, fail_num
+            m = re.match(p, line)
+            if m is not None:
+                shutil.copyfile(m.group(1), os.path.join(os.path.split(result)[0], xml_file_name))
+                total_num = int(m.group(2))
+                success_num = int(m.group(3))
+    return total_num, success_num
 
 
-def exec_am_cmd(am_cmd, device, target_package, result_file, logcat_file):
+def exec_cmd(cmd, device, target_package, result_file, logcat_file, test_path):
```

Comment:
i assume that `test_path` is a path that the `cmd` will be executed.
if i am right, then `test_path` is not an appropriate name

Reply:
done